schemas/telemetry/frecency-update/frecency-update.4.parquetmr.txt

```diff
@@ -3,21 +3,32 @@ message frecency-update {
   required binary clientId (UTF8);
   required binary type (UTF8);
   required group payload {
+    required int32 model_version;
     required group frecency_scores (LIST) {
       repeated group list {
         required float element;
       }
     }
-    required int32 model_version;
     required float loss;
-    required int32 num_chars_typed;
-    required int32 rank_selected;
-    required int32 num_suggestions_displayed;
-    required binary study_variation;
     required group update (LIST) {
       repeated group list {
         required float element;
       }
     }
+    required int32 num_suggestions_displayed;
+    required int32 rank_selected;
+    required int32 bookmark_and_history_num_suggestions_displayed;
+    required int32 bookmark_and_history_rank_selected;
+    required int32 num_key_down_events_at_selecteds_first_entry;
+    required int32 num_key_down_events;
+    required int32 time_start_interaction;
+    required int32 time_end_interaction;
+    required int32 time_at_selecteds_first_entry;
+    required int32 search_string_length;
+    required binary selected_style;
+    required int32 selected_url_was_same_as_search_string;
+    required int32 enter_was_pressed;
+    required binary study_variation;
+    required binary study_addon_version;
```

Comment:
I don't see an identifier field for the experiment instance – is `study_addon_version` a proxy for that? I suspect an explicit experiment identifier would be useful regardless (since this ping doesn't include the environment we won't have an `experiments` field or any other explicit experiment identifier, and we're still receiving a handful of pings every day from the old experiment :/)

Reply:
Indeed the version number is a proxy for that in this schema, which should be fine since the `frecency-update` is only consumed used meanwhile the study is running and we don't plan to run several studies using this schema.
(In v2 of the add-on, [`shield-study-addon` pings are being sent alongside the `frecency-update` ping](https://github.com/mozilla/federated-learning-v2-study-addon/blob/master/docs/TELEMETRY.md#shield-study-addon-pings-specific-to-this-study), which contains the environment and the shield envelope with the add-on id etc and is used for post-study analysis.)
As for the v1 pings, the overwrite of the previous schema in this PR will make the old pings get dropped.

Reply:
I would strongly consider adding an experiment identifier in any case for semantic clarity (a future analyst comparing some of these metrics across different experiment instances would need to look up the addon version that was in use for that particular experiment, for instance), but this PR is about correctness, not semantics (and I note it appears you've gotten the addon signed so I assume you're not interested in making changes to the addon 😉)